extension/src/experiments/processExecution/index.ts

```diff
@@ -1,23 +1,23 @@
-import { join } from 'path'
-import { EXP_RWLOCK_FILE } from '../../cli/dvc/constants'
-import { getPidFromFile } from '../../fileSystem'
-import { Toast } from '../../vscode/toast'
+import { collectDvcRootPids } from './collect'
 import { processExists, stopProcesses } from '../../process/execution'
 
-export const stopWorkspaceExperiment = async (dvcRoot: string) => {
-  const pid = await getPidFromFile(join(dvcRoot, EXP_RWLOCK_FILE))
-  if (!pid) {
-    return
-  }
-  if (!(await processExists(pid))) {
-    return
+export const stopWorkspaceExperiment = async (dvcRoot: string, id: string) => {
+  const pids = new Set<number>()
+
+  await collectDvcRootPids(pids, dvcRoot)
+
+  const [pid] = [...pids]
```

Comment:
How about `const pid = pids.slice(0, 1)`? It doesn't matter one or the other, I guess it's just a preference, but to me it looks like 2 steps vs. one.

Reply:
I would still have to convert the set into an array (`error TS2339: Property 'slice' does not exist on type 'Set<number>'`). Whatever I do it will look a bit whacky. This comment did make me realise that I needed another logic branch "to be safe". Thanks

Reply:
I get it now. I thought you were avoiding mutation, I didn't follow to see it was a `Set`, my bad. Thanks for explaining.